src/Build/BackEnd/Components/RequestBuilder/TargetBuilder.cs

```diff
@@ -777,6 +777,12 @@ private void ComputeAfterTargetFailures(string[] targetNames)
                     var targetsToCheckForAfterTargets = new Queue<string>();
                     targetsToCheckForAfterTargets.Enqueue(targetName);
 
+                    // Set of targets already processed, to break cycles of AfterTargets
+                    var targetsChecked = new HashSet<string>(MSBuildNameIgnoreCaseComparer.Default)
```

Comment:
nit: The `HashSet` could be constructed lazily when first needed inside the `foreach` below. It would eliminate the allocation in cases where the target has no AfterTargets (true for most targets?) I believe perf-wise it's still OK the way it is now as this is not a super hot path. Thank you!